include/deal.II/non_matching/mapping_info.h

```diff
@@ -157,30 +293,550 @@ namespace NonMatching
   void
   MappingInfo<dim, spacedim>::reinit(
     const typename Triangulation<dim, spacedim>::cell_iterator &cell,
-    const ArrayView<const Point<dim>> &                         unit_points)
+    const ArrayView<const Point<dim>> &                         unit_points_in)
+  {
+    unit_points =
+      std::vector<Point<dim>>(unit_points_in.begin(), unit_points_in.end());
+
+    mapping_data.resize(1);
+    compute_mapping_data_for_generic_points(cell,
+                                            unit_points_in,
+                                            mapping_data[0]);
+
+    state = State::single_cell;
+  }
+
+
+  template <int dim, int spacedim>
+  void
+  MappingInfo<dim, spacedim>::reinit(
+    const typename Triangulation<dim, spacedim>::cell_iterator &cell,
+    const Quadrature<dim> &                                     quadrature)
+  {
+    const auto &points  = quadrature.get_points();
+    const auto &weights = quadrature.get_weights();
+
+    reinit(cell, points);
+
+    if (update_flags_mapping & update_JxW_values)
+      for (unsigned int q = 0; q < points.size(); ++q)
+        mapping_data[0].JxW_values[q] =
+          determinant(Tensor<2, dim>(mapping_data[0].jacobians[q])) *
+          weights[q];
+  }
+
+
+  template <int dim, int spacedim>
+  template <typename Iterator>
+  void
+  MappingInfo<dim, spacedim>::reinit_cells(
+    const IteratorRange<Iterator> &             cell_iterator_range,
+    const std::vector<std::vector<Point<dim>>> &unit_points_vector,
+    const unsigned int                          n_unfiltered_cells)
+  {
+    do_cell_index_compression =
+      n_unfiltered_cells != numbers::invalid_unsigned_int;
+
+    const unsigned int n_cells =
+      std::distance(cell_iterator_range.begin(), cell_iterator_range.end());
+
+    Assert(n_cells == unit_points_vector.size(),
+           ExcDimensionMismatch(n_cells, unit_points_vector.size()));
+
+    // fill unit points index offset vector
+    unit_points_index.resize(n_cells + 1);
+    unsigned int cell_index    = 0;
+    unsigned int n_unit_points = 0;
+    for (const auto &cell : cell_iterator_range)
+      {
+        unit_points_index[cell_index] = n_unit_points;
+        n_unit_points += unit_points_vector[cell_index].size();
+
+        ++cell_index;
+      }
+    unit_points_index[n_cells] = n_unit_points;
```

Comment:
Also here, I would prefer to loop over `unit_points_vector` instead. Maybe like this, if you would like to keep the range-based for loop:
```suggestion
    unit_points_index.reserve(n_cells + 1);
    unit_points_index.resize(1);
    unit_points_index[0] = 0;
    for (const auto &unit_points : unit_points_vector)
      unit_points_index.push_back(unit_points_index.back() + unit_points.size());
```

Reply:
I have replaced it with code similar to your suggestion for the cellwise `reinit_*` functions. For the faces I could not think of a better way because I need access to `cell->face_indices()` through the `CellIterator`.